lib/services/role.go

```diff
@@ -2515,6 +2546,51 @@ func (set RoleSet) CheckAccessToRule(ctx RuleContext, namespace string, resource
 	})
 }
 
+// GetKubeResources returns allowed and denied list of Kubernetes Resources configured in the RoleSet.
+func (set RoleSet) GetKubeResources(cluster types.KubeCluster) (allowed, denied []types.KubernetesResource) {
+	var (
+		allowedResources []types.KubernetesResource
+		deniedResources  []types.KubernetesResource
+	)
+	getRoleLabels := types.Role.GetKubernetesLabels
+	for _, role := range set {
+		matchLabels, _, err := MatchLabels(getRoleLabels(role, types.Allow), cluster.GetAllLabels())
+		if err != nil {
+			continue
+		}
+		if !matchLabels {
+			continue
+		}
+		allowedResources = append(allowedResources, role.GetKubeResources(types.Allow)...)
+	}
+
+	for _, role := range set {
+		matchLabels, _, err := MatchLabels(getRoleLabels(role, types.Deny), cluster.GetAllLabels())
+		if err != nil {
+			continue
+		}
+		if !matchLabels {
+			continue
+		}
+		deniedResources = append(deniedResources, role.GetKubeResources(types.Deny)...)
+	}
+
+	return deduplicateKubeResources(allowedResources), deduplicateKubeResources(deniedResources)
+}
+
+func deduplicateKubeResources(sliceList []types.KubernetesResource) []types.KubernetesResource {
+	allKeys := make(map[string]bool)
```

Comment:
A `map[string]struct{}` makes for a good set (struct{} uses no additional space). A nice to have, but no need to change.